src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Flows/AcquireTokenByAuthorizationCodeHandler.cs

```diff
@@ -69,9 +70,9 @@ protected override void AddAditionalRequestParameters(DictionaryRequestParameter
             requestParameters[OAuthParameter.RedirectUri] = this.redirectUri.OriginalString;
         }
 
-        protected override void PostTokenRequest(AuthenticationResultEx resultEx)
+        protected override async Task PostTokenRequest(AuthenticationResultEx resultEx)
```

Comment:
UpdateFromTemplateAsync method is doing authority validation by sending instance discovery http request